README.md

```diff
@@ -68,18 +68,23 @@ default location for staging temporary files to the /tmp directory.
 The normal default is /shm. /shm is a RAM disk which defaults to a 64 MB size
 in Docker containers, too small for typical scientific computing.
 
+# Install a new PyPi Package Dependency
+1. Create a new python environment `python  -m venv .venv`.
+2. Activate your new python environment `source .venv/bin/activate`
+3. Add new dependency to `requirements-core.txt` file. Include the version
+4. Run script `generate-requirements-full.sh`
+
+# Updating existing PyPi Package Version
+1. Create a new python environment `python  -m venv .venv`.
+2. Activate your new python environment `source .venv/bin/activate`
+3. Update version of existing package in requirements-core.txt
+4. Run script `generate-requirements-full.sh`
```

Comment:
pip-tools doesn't need a virtual environment. When the `pip-compile` command runs, it internally creates its own temporary virtual env and uses that. Whatever environment the developer is already in and whatever packages are installed in it has no effect to the final `requirements-full.txt` (which is neat!).

Also, do you think we should _discourage_ folks from adding new packages to `requirements-core.txt` moving forward? Maybe we should include language about that somewhere in this README. If possible, we should keep this image as lean as possible, and ask users to just `pip install <whatever>` they might want should they want other packages in their use case.

```suggestion
# Install a new PyPi Package Dependency
1. Add new dependency to `requirements-core.txt` file. Include the version
2. Run script `generate-requirements-full.sh`

# Updating existing PyPi Package Version
1. Update version of existing package in `requirements-core.txt`
2. Run script `generate-requirements-full.sh`
```

Reply:
That is a great question.

I think it depends on 2 things highlighted by this Truth Table

1. How many uses cases will be using the new package - number of projects 
    1 = a lot of projects
    0 = a small number of projects

2. The frequency at which the new package will be used 
   1 = used a lot
   0 = not used a lot

        1 2 | 
        ----------
    A  1 1  | 1  -  many projects uses package frequently per hour
    B  1 0 | 1  -  many projects uses package but not many runs of these projects per hour
    C  0 1 | 1  - not many projects uses package, but those that do run frequently per hour
    D  0 0 | 0 - not many projects uses package and do not run frequently per hour.
    
    
While easy to conceptualize, it is probably difficult to practically answer. Generally speaking, if the cost of having the package already installed on the ds-python image outweighs the cost of having to pip install the dependency every time, then i think we should encourage engineers to add the dependency here.
if not, then no.